torch_geometric/contrib/flag/flag.py

```diff
@@ -81,15 +84,21 @@ def forward(
                 device=self.device,
             ).uniform_(-step_size, step_size))
 
+        [c.on_ascent_step_begin(0, None) for c in self.callbacks]
+
         # out = forward(perturb)
         out = self.model(x + self.perturb, edge_index)[train_idx]
         self.loss = self.loss_fn(out, training_labels)
 
         # loss /= args.m
         self.loss /= n_ascent_steps
 
+        [c.on_ascent_step_end(0, self.loss, self.perturb) for c in self.callbacks]
+
         # for _ in range(args.m - 1):
-        for _ in range(n_ascent_steps - 1):
+        for i in range(n_ascent_steps - 1):
+            [c.on_ascent_step_begin(i+1, self.loss) for c in self.callbacks]
```

Comment:
super nit: `i + 1` instead of `i+1`. I'm surprised flake8 didn't complain at you!

Reply:
haha ty!